### PR TITLE
key rotator: implement basic Key data type & rotation logic.

### PR DESCRIPTION
**Pull request by BranLwyd**
I'm happy with most of this. The one thing I may consider changing as I move forward with implementation is the RotationConfig.CreateKeyFunc; I may factor this out to something like a "key profile" (name not final) that knows how to generate new keys, serialize them for the various places they need to be written, etc -- the idea being I'd have one key profile for batch signing keys & one key profile for packet encryption keys.

### CODE REVIEW

**Review by tgeoghegan — CHANGES_REQUESTED** [review]
I think you will need to add job definitions in `.github/workflows/ci-build.yml` to build and test `key-rotator`.

**Comment by tgeoghegan** [comment]
I see now that my concerns about the type returned by `CreateKeyFunc` and `Version.KeyMaterial` are addressed by the `Raw` type introduced in #1081, so feel free to ignore that feedback.

File: key-rotator/key/key.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	return Key(kvs), nil
+}
+
+// Version represents a single version of a key, i.e. raw private key material,
+// as well as associated metadata. Typically, a Version will be embedded within
+// a Set.
+type Version struct {
+	KeyMaterial  string    `json:"key,omitempty"`
+	CreationTime time.Time `json:"creation_time,omitempty"`
+	Primary      bool      `json:"priamry,omitempty"`
```
[the remark this answers is not in the document]
Comment (BranLwyd) [reply key-rotator/key/key.go:149]: Done.

**Review by tgeoghegan — APPROVED** [review]
Looks good! I think this may conflict with #1066, though, or at least you'll want to make sure you configure a golang-ci job for key-rotator.

**Pull request merged**